apps/dashboard/app/models/user_configuration.rb

```diff
@@ -86,22 +87,6 @@ def navbar_type
     end
   end
 
-  # What to group pinned apps by
-  # @return [String, ""] Defaults to ""
-  def pinned_apps_group_by
-    group_by = ENV['OOD_PINNED_APPS_GROUP_BY'] || fetch(:pinned_apps_group_by, '')
-
-    # FIXME: the user_configuration shouldn't really know the API of
-    # OodApp or subclasses. This is a hack because subclasses of OodApp overload
-    # the category and subcategory to something new while saving the original.
-    # The fix would be to move this knowledge to somewhere more appropriate than here.
-    if group_by == 'category' || group_by == 'subcategory'
-      "original_#{group_by}"
-    else
-      group_by
-    end
```

Comment:
This is the main driver for this pull request. Things like UserConfiguration having this knowledge of re-categorization.